tools/justfile

```
@@ -0,0 +1,144 @@
+# Default values for the deployment
+tag    := "test"
+queue  := "active"
+worker := "htcmock"
+
+# Base compose file
+export COMPOSE_BASE := "../docker-compose/docker-compose.yml"
+
+# loggin level of deployment
+export LOGGING_LEVEL:="Information"
+
+# Sets the queue
+export QUEUE := if queue == "rabbit" {
+  "../docker-compose/docker-compose.queue-rabbitmq.yml"
+}else if  queue == "rabbit091" {
+  "../docker-compose/docker-compose.queue-rabbitmq.yml"
+}else if queue == "artemis" {
+  "../docker-compose/docker-compose.queue-artemis.yml"
+}else{
+  "../docker-compose/docker-compose.queue-activemqp.yml"
+}
+
+
+# Sets the override to feed docker-compose
+export OVERRIDE := if  queue == "rabbit091" {
+  "../docker-compose/docker-compose.override-rabbitmq091.yml"
+}else{
+  "../docker-compose/docker-compose.override.yml"
+}
+
+# Defines worker and enviroment variables for deployment
+export ARMONIK_WORKER:= if worker == "stream" {
+  "dockerhubaneo/armonik_core_stream_test_worker:"+tag
+}else{
+  "dockerhubaneo/armonik_core_htcmock_test_worker:"+tag
+}
+export ARMONIK_METRICS:="dockerhubaneo/armonik_control_metrics:"+tag
+export ARMONIK_PARTITIONMETRICS:="dockerhubaneo/armonik_control_partitionmetrics:"+tag
+export ARMONIK_SUBMITTER:="dockerhubaneo/armonik_control:"+tag
+export ARMONIK_POLLINGAGENT:="dockerhubaneo/armonik_pollingagent:"+tag
+
+# List recipes and their usage
+@default:
+  just --list
+  just _usage
+
+_usage:
+  #!/usr/bin/env bash
+  set -euo pipefail
+  echo
+  echo "The recipe deploy takes three arguments"
+  echo
+  echo "usage: just tag=<tag> queue=<queue> worker=<worker> deploy"
+  echo "       if any of the variables is not set, its default value is used"
+  echo
+  echo "   tag: The core tag image to use, defaults to test"
+  echo
+  echo "   queue: allowed values below"
+  cat <<-EOF
+        active   :  for activemq (1.0.0 protocol) (default)
+        rabbit   :  for rabbitmq (1.0.0 protocol)
+        rabbit091:  for rabbitmq (0.9.1 protocol)
+        artemis  :  for artemismq (1.0.0 protocol)
+  EOF
+  echo
+  echo "   worker: allowed values below"
+  cat <<-EOF
+        htcmock: for HtcMock V3 (default)
+        stream: for Stream worker
+  EOF
+
+# Custom command to build a single image
+_build-image imageTag dockerFile:
+  docker build -t {{ imageTag }} -f {{ dockerFile }} ../
+
+# Build all images necessary for the deployment 
+_build-armonik-components:
```

rename the rule to `build-all`?
I also think you can call subrule at the right of the colon, instead of calling `just` inside.
```suggestion
_build-armonik-components: (build ARMONIK_WORKER "../Tests/Stream/Server/Dockerfile") (build ARMONIK_METRICS "../Control/Metrics/src/Dockerfile")...
```

---

Agree with the renaming, regarding subruling after the colon I find more readable to call `just` it inside.

---

fair enough